### PR TITLE
x11-misc/dunst: Update USE flags

### PR DESCRIPTION
**Pull request by fedang**
Newer version of dunst support conditional compilation of certain features, like the X backend.

dunst-1.11.0-r1.ebuild and dunst-9999.ebuild were modified to reflect that.

NOTE: I made a revision because the RDEPEND changed

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: x11-misc/dunst/dunst-9999.ebuild [hunk 1]
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
 
 LICENSE="BSD"
 SLOT="0"
-IUSE="wayland"
+IUSE="+X wayland +dunstify +completions +xdg"
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-9999.ebuild:15]: Please sort alphabetically.

File: x11-misc/dunst/dunst-1.11.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+inherit systemd toolchain-funcs
+
+DESCRIPTION="Lightweight replacement for common notification daemons"
+HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
+SRC_URI="https://github.com/dunst-project/dunst/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+X wayland +dunstify +completions +xdg"
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-1.11.0-r1.ebuild:15]: Please sort alphabetically.

File: x11-misc/dunst/dunst-9999.ebuild [hunk 3]
```diff
@@ -70,6 +79,9 @@ src_install() {
 		SYSCONFDIR="${ED}/etc/xdg"
 		SYSTEMD="0"
 		WAYLAND="$(usex wayland 1 0)"
+		X11="$(usex X 1 0)"
+		DUNSTIFY="$(usex dunstify 1 0)"
+		COMPLETIONS="$(usex completions 1 0)"
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-9999.ebuild:84]: Please sort here also alphabetically.

File: x11-misc/dunst/dunst-9999.ebuild [hunk 4]
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
...
 LICENSE="BSD"
 SLOT="0"
-IUSE="wayland"
+IUSE="+X wayland +dunstify +completions +xdg"
 
 DEPEND="
 	dev-libs/glib:2
 	sys-apps/dbus
-	x11-libs/cairo[X,glib]
+	x11-libs/cairo[glib]
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-9999.ebuild:20]: Instead adding ```x11-libs/cairo[X]``` to ```X? (``` you could do instead: ```x11-libs/cairo[X?,glib]```, so you don't have to add twice the same dep with different use flags.

File: x11-misc/dunst/dunst-1.11.0-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+src_install() {
+	local myemakeargs=(
+		PREFIX="${ED}/usr"
+		SYSCONFDIR="${ED}/etc/xdg"
+		SYSTEMD="0"
+		WAYLAND="$(usex wayland 1 0)"
+		X11="$(usex X 1 0)"
+		DUNSTIFY="$(usex dunstify 1 0)"
+		COMPLETIONS="$(usex completions 1 0)"
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-1.11.0-r1.ebuild:84]: Please also sort here alphabetically.

File: x11-misc/dunst/dunst-9999.ebuild [hunk 6]
```diff
@@ -12,21 +12,25 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
...
 	x11-libs/gdk-pixbuf:2
-	x11-libs/libX11
-	x11-libs/libXext
-	x11-libs/libXScrnSaver
-	x11-libs/libXinerama
-	x11-libs/libXrandr
-	x11-libs/libnotify
-	x11-libs/pango[X]
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-9999.ebuild:28]: Same as obove for cairo.

File: x11-misc/dunst/dunst-1.11.0-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+X wayland +dunstify +completions +xdg"
+
+DEPEND="
+	dev-libs/glib:2
+	sys-apps/dbus
+	x11-libs/cairo[glib]
+	x11-libs/gdk-pixbuf:2
+	x11-libs/pango
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-1.11.0-r1.ebuild:22]: Same as obove for cairo.

File: x11-misc/dunst/dunst-1.11.0-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+IUSE="+X wayland +dunstify +completions +xdg"
+
+DEPEND="
+	dev-libs/glib:2
+	sys-apps/dbus
+	x11-libs/cairo[glib]
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-1.11.0-r1.ebuild:20]: Instead adding ```x11-libs/cairo[X]``` to ```X? (``` you could do instead: ```x11-libs/cairo[X?,glib]```, so you don't have to add twice the same dep with different use flags.

**Review by ConiKost — CHANGES_REQUESTED** [review]
Left some feedback :-)

File: x11-misc/dunst/dunst-9999.ebuild (hunk 4 above)
Comment (fedang) [reply x11-misc/dunst/dunst-9999.ebuild:20]: thanks, didn't know about that :+1:

**Comment by fedang** [comment]
> Left some feedback :-)

Thank you, this was my first ebuild.
Is it okay now?

File: x11-misc/dunst/metadata.xml [hunk 9]
```diff
@@ -5,6 +5,11 @@
 		<email>conikost@gentoo.org</email>
 		<name>Conrad Kostecki</name>
 	</maintainer>
+	<use>
+		<flag name="dunstify">Build dunstify (notify-send alternative)</flag>
+		<flag name="completions">Install shell completions (for bash, zsh and fish)</flag>
```
Comment (ConiKost) [inline x11-misc/dunst/metadata.xml:10]: Also here please alphabetically.

**Comment by fedang** [comment]
done :+1:

File: x11-misc/dunst/dunst-9999.ebuild [hunk 10]
```diff
@@ -12,21 +12,23 @@ HOMEPAGE="https://dunst-project.org/ https://github.com/dunst-project/dunst"
...
+	x11-libs/pango[X?]
+	xdg? ( x11-misc/xdg-utils )
+	X? (
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXScrnSaver
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+	)
+	dunstify? ( x11-libs/libnotify )
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-9999.ebuild:31]: Please look again through, this is also not sorted.

Comment (fedang) [reply x11-misc/dunst/dunst-9999.ebuild]: ok done it.
is there an ebuild style guide so I don't have to make such mistakes?

**Comment by fedang** [comment]
I have reordered the depends.

Also I wanted to mention this. Now dunst does not required X anymore.
Since it equally supports wayland and x11, maybe it would be more appropriate to be in app-misc or gui-apps (like mako), rather than x11-misc?
But this is just a suggestion, I don't know the policy on changing categories

File: x11-misc/dunst/metadata.xml [hunk 11]
```diff
@@ -5,6 +5,11 @@
 		<email>conikost@gentoo.org</email>
 		<name>Conrad Kostecki</name>
 	</maintainer>
+	<use>
+		<flag name="completions">Install shell completions (for bash, zsh and fish)</flag>
```
Comment (ConiKost) [inline x11-misc/dunst/metadata.xml:9]: One last small nit pick. Please order ```bash, fish and zsh```. Thanks! Otherwise lgtm now.

Comment (fedang) [reply x11-misc/dunst/metadata.xml]: done :+1:

**Comment by ConiKost** [comment]
Yes, I think its worth to add it. You can also drop 1.11.0-r1 commits, given that there is now a new release, I would tend to add you changes about use flags etc. only to newest 1.12.0 version.

**Comment by fedang** [comment]
I removed 1.11.0-r1.

Also added in `pkg_postinst` a message about the breaking change.
I did not include the full message for length. But if you prefer the entire thing let me know.

File: x11-misc/dunst/dunst-1.12.0.ebuild [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+		X11="$(usex X 1 0)"
+	)
+
+	emake "${myemakeargs[@]}" install
+
+	systemd_newuserunit dunst.systemd.service.in dunst.service
+}
+
+pkg_postinst() {
+	einfo "The behaviour of the setting 'height' has been changed in a breaking way."
```
Comment (ConiKost) [inline x11-misc/dunst/dunst-1.12.0.ebuild:91]: While the text is fine, this should be only display for users, who are upgrading. See, for example, ```net-im/prosody```, how you could do that.

Comment (fedang) [reply x11-misc/dunst/dunst-1.12.0.ebuild:91]: 👍

**Comment by fedang** [comment]
@ConiKost I just added a check for the REPLACING_VERSIONS.
I think patching the user config is not advisable

**Review by ConiKost — CHANGES_REQUESTED** [review]
thanks! lgtm. But please squash your commits.

**Comment by fedang** [comment]
> thanks! lgtm. But please squash your commits.

I squashed to a single commit

**Review by ConiKost — APPROVED** [review]

**Pull request closed without merging**